### ISSUE TITLE
Add line ending setting to code editor preferences

### ISSUE THREAD

**Issue by dvatp** [octobercms/october#1612]
October's backend code editor saves files with CR/LF (windows) line endings, and my installation lives on Unix.

I occasionally edit files manually in my favorite editor and cannot stand seeing a ^M at the end of every line.  I'd strip the CR manually but if I have to tweak something quickly and use the backend code editor it just reformats it again.  Very frustrating.

I looked around the backend for an editor preference to change this but there isn't one.  Can we add one?  I want to select LF for my installation.

**Comment by Abdillah** [comment]
I'm not found any code which producing `\r\n` anywhere (with simple grep and GitHub search). Then, Is this related to specification written on [this comment]?
Thus, option addition means we add function which replace `\r\n` to `\n`?

[this comment]: (https://github.com/ajaxorg/ace/issues/1515#issuecomment-20971401)

**Comment by daftspunk** [comment]
Check the `config/cms.php` file

```

    /*
    |--------------------------------------------------------------------------
    | Convert Line Endings
    |--------------------------------------------------------------------------
    |
    | Determines if October should convert line endings from the windows style
    | \r\n to the unix style \n.
    |
    */

    'convertLineEndings' => false,
```

**Comment by Abdillah** [comment]
Yes! I've just found that option. And want to share it here... Thanks @daftspunk .
But, this one only convert the markup as shown by [the code](https://github.com/octobercms/october/blob/master/modules/cms/controllers/Index.php#L158) and my experiment.

**Comment by daftspunk** [comment]
Fixed by #1890
